### PR TITLE
feat(FX-4609, FX-4590): preparation

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feat**

This PR solves [FX-4609], [FX-4590]

### Description
In the future we should display [the manage artwork modal](https://www.figma.com/file/wSV2KNz97Tyj8HkH2ZYAkB/Save-%2B-Wishlist?node-id=308%3A123937&t=4lZWhJ77zmlGTUoi-0) when the 💙 icon is clicked on the saves/wishlist item. These are the changes that will come in handy for this

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4609]: https://artsyproduct.atlassian.net/browse/FX-4609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4590]: https://artsyproduct.atlassian.net/browse/FX-4590?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Artwork/SaveButton/NewSaveButton.tsx [hunk 1]
```diff
@@ -17,38 +17,20 @@ export interface SaveButtonProps {
...
+  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+}
+
 const BTN_HEIGHT = 20
 const BTN_WIDTH = 20
 
-export const NewSaveButton: React.FC<SaveButtonProps> = ({
-  artwork,
-  contextModule,
+export const NewSaveButtonBase: React.FC<NewSaveButtonBaseProps> = ({
```
Comment (damassi) [inline src/Components/Artwork/SaveButton/NewSaveButton.tsx:28]: #non-blocking I know this code was already roughly here so no need to change in this PR, but is there any way we can avoid using "New" in the name, or rename old `SaveButton` to `DeprecatedSaveButton` and then this to `SaveButton`?

**Review by damassi — APPROVED** [review]

**Review by gkartalis — APPROVED** [review]
noice! 🌟⭐🌟

File: src/Components/Artwork/SaveButton/NewSaveButton.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/Components/Artwork/SaveButton/NewSaveButton.tsx:28]: Good suggestion. I'll rename it

Comment (damassi) [reply src/Components/Artwork/SaveButton/NewSaveButton.tsx:28]: oh yeah, no prob at all! tagged it non-blocking but wanted to call it out 👍

Comment (dimatretyak) [reply src/Components/Artwork/SaveButton/NewSaveButton.tsx:28]: thanks!

Comment (dimatretyak) [reply src/Components/Artwork/SaveButton/NewSaveButton.tsx:28]: Addressed here: https://github.com/artsy/force/pull/11932

**Pull request merged**